tests/e2e/stablehlo_ops/fft_complex.mlir

```diff
@@ -0,0 +1,444 @@
+  return
+}
+
+// 1D RFFT.
+func.func @rfft_1d() {
+  %input = util.unfoldable_constant dense<[0.7555577, 0.66894025, 0.36586154, 0.24478266, 0.7804641, 0.9256327, 0.42047995, 0.44436365, 0.9899847, 0.97323716, 0.21359204, 0.23445652, 0.38268244, 0.9832261, 0.81922436, 0.033826284, 0.87232935, 0.9702218, 0.795888, 0.05799015, 0.22904389, 0.42274877, 0.70018625, 0.46720484, 0.04191081, 0.26426974, 0.123912655, 0.6813819, 0.78704375, 0.16205642, 0.33862734, 0.43718913]> : tensor<32xf32>
+  %0 = stablehlo.fft %input, type = RFFT, length = [32] : (tensor<32xf32>) -> tensor<17xcomplex<f32>>
+  %1 = stablehlo.real %0 : (tensor<17xcomplex<f32>>) -> tensor<17xf32>
+  %2 = stablehlo.imag %0 : (tensor<17xcomplex<f32>>) -> tensor<17xf32>
+  check.expect_almost_eq_const(%1, dense<[16.588318, -0.46602482, 0.8568451, -1.1161734, 0.86731994, -0.5296755, 0.7260204, 3.421573, 1.0612442, -0.33810878, -0.7522839, -1.0411217, 0.09377679, -0.48347437, 1.5533843, -0.38116747, 0.6452608]> : tensor<17xf32>) : tensor<17xf32>
+  check.expect_almost_eq_const(%2, dense<[0.0, -2.0378456, -0.33985636, 0.6871074, 0.62438464, -0.90058994, -1.6517396, 2.8402321, -2.7691374, 0.4236927, -0.21823317, 1.0264674, -0.9341428, -0.76218516, 0.45277753, -0.24614418, 0.0]> : tensor<17xf32>) : tensor<17xf32>
+  return
+}
+
+// 1D IRFFT. This is the inverse of the @rfft_1d() so should give back the original inputs (within floating point error).
+func.func @irfft_1d() {
+  %input = util.unfoldable_constant dense<[(16.588318,0.0), (-0.46602482,-2.0378456), (0.8568451,-0.33985636), (-1.1161734,0.6871074), (0.86731994,0.62438464), (-0.5296755,-0.90058994), (0.7260204,-1.6517396), (3.421573,2.8402321), (1.0612442,-2.7691374), (-0.33810878,0.4236927), (-0.7522839,-0.21823317), (-1.0411217,1.0264674), (0.09377679,-0.9341428), (-0.48347437,-0.76218516), (1.5533843,0.45277753), (-0.38116747,-0.24614418), (0.6452608,0.0)]> : tensor<17xcomplex<f32>>
+  %0 = stablehlo.fft %input, type = IRFFT, length = [32] : (tensor<17xcomplex<f32>>) -> tensor<32xf32>
+  check.expect_almost_eq_const(%0, dense<[0.75555766, 0.6689402, 0.3658616, 0.24478275, 0.78046405, 0.9256327, 0.42048004, 0.4443637, 0.98998463, 0.97323704, 0.21359204, 0.2344566, 0.38268244, 0.9832261, 0.81922436, 0.033826366, 0.87232935, 0.97022176, 0.795888, 0.05799026, 0.22904384, 0.42274877, 0.7001864, 0.46720487, 0.041910768, 0.26426968, 0.12391268, 0.6813819, 0.7870437, 0.16205639, 0.33862743, 0.43718916]> : tensor<32xf32>) : tensor<32xf32>
+  return
+}
+
+// 1D FFT on the innermost dimension of a 2D tensor.
+func.func @fft_2d() {
+  %input = util.unfoldable_constant dense<[
```

Comment:
The tests before this seem fine to add here, but tests from here and below seem pretty large constants to add this way. I wonder if you can just use Torch.fft to check for correctness and add these tests to https://github.com/iree-org/iree-test-suites/tree/main/torch_ops . An even better solution would be to maybe have a test suite that is stablehlo based, and is setup similar to how the torch ops test suite is setup.